spec/support/concerns/job_transactionable_shared_examples.rb

```diff
@@ -0,0 +1,58 @@
+shared_examples 'a job_transactionable model' do
+  it {
+    expect(described_class).to include(JobTransactionable)
+    is_expected.to respond_to(:current_transaction)
+    is_expected.to have_many(:job_transactions)
+
+    is_expected.to respond_to(:root_create_transaction)
+    is_expected.to respond_to(:root_update_transaction)
+    is_expected.to callback(:root_create_transaction)
+      .before(:create)
+    is_expected.to callback(:root_update_transaction)
+      .before(:update)
+  }
+
+  describe '.create_transaction' do
+    let(:transaction_state) {'testing'}
+
+    context 'with nil current_transaction' do
+      it {
```

Comment:
It would be good to test that `#current_transaction` is nil before calling `#create_transaction`